test/odbc/psqlodbc/psqlodbc_tests_common.h

```diff
@@ -0,0 +1,696 @@
+
+void testCommonCharColumnAttributes(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, vector<int> lengthExpected, vector<int> precisionExpected, vector<int> scaleExpected, vector<string> nameExpected, vector<int> caseSensitivityExpected, vector<string> prefixExpected, vector<string> suffixExpected);
+
+void testTableCreationFailure(string tableName, vector<vector<pair<string, string>>> invalidColumns);
+
+void testInsertionSuccessChar(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, vector<string> insertedValues, vector<string> expectedInsertedValues);
+
+void testInsertionFailure(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, vector<string> invalidInsertedValues, bool isNumericInsert);
+
+void testUpdateSuccessChar(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, string colNameToUpdate, vector<string> insertedValues, vector<string> expectedInsertedValues, vector<string> updatedValues, vector<string> expectedUpdatedValues);
+
+void testUpdateFailChar(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, string colNameToUpdate, vector<string> insertedValues, vector<string> expectedInsertedValues, vector<string> updatedValues);
+
+void testViewCreationChar(string viewName, string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, vector<string> insertedValues, vector<string> expectedInsertedValues);
+
+void testPrimaryKeysChar(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, string schemaName, string pkTableName, vector<string> primaryKeyColumns, vector<string> insertedValues, vector<string> expectedInsertedValues);
+
+void testUniqueConstraintChar(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, vector<string> uniqueConstraintColumns, vector<string> insertedValues, vector<string> expectedInsertedValues);
+
+void testComparisonOperators(string tableName, vector<pair<string, string>> tableColumns, string col1Name, string col2Name, vector<string> col1Data, vector<string> col2Data, vector<string> operationsQuery, vector<vector<char>> expectedResults);
+
+void testComparisonFunctionsChar(string tableName, vector<pair<string, string>> tableColumns, vector<string> insertedData, vector<string> operationsQuery, vector<string> expectedResults);
+
+template <typename T>
+void testInsertionSuccessNumeric(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, int type, T data, int bufferLen, vector<string> insertedValues, vector<T> expectedInsertedValues, vector<long> expectedLen) {
+  
+  OdbcHandler odbcHandler(Drivers::GetDriver(ServerType::PSQL));
```

Comment:
ServerType::PSQL
Is this code only applicable to postgres connections? Would we use different ODBC calls/logic for SQL Server?
If it would be the same for SQL Server it would be better to pass ServerType as parameter or have some function or variable where the current ServerType could be accesse.

Even though right now we're not using the refactored functions for SQL Server tests, it would be nice if they could be used in the future if needed.

Reply:
I agree with this thought. I now have `ServerType` as part of the function parameters.